sky/jobs/controller.py

```diff
@@ -948,6 +953,40 @@ async def run_job_loop(self,
                             ctx.override_envs({key: value})
                             logger.debug('Set environment variable: %s=%s', key,
                                          value)
+
+                    # Set SKYPILOT_JOB_RANK from job_id_to_rank mapping if
+                    # available.
+                    if ('SKYPILOT_JOB_ID_TO_RANK' in env_vars and
+                            env_vars['SKYPILOT_JOB_ID_TO_RANK']):
+                        try:
+                            job_id_to_rank = (json.loads(
+                                env_vars['SKYPILOT_JOB_ID_TO_RANK']))
+                            logger.debug('Loaded job_id_to_rank mapping: %s',
```

Comment:
style nit: Prefer f-strings. I think there are a few stray cases of %s formatting in this file that we need to remove and the AI agents seem to love adding more :/